components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/connector/recaptcha/EmailOTPCaptchaConnector.java

```diff
@@ -0,0 +1,116 @@
+
+/**
+ * Recaptcha Connector for Email OTP
+ */
+public class EmailOTPCaptchaConnector extends AbstractReCaptchaConnector {
+    private static final Log log = LogFactory.getLog(EmailOTPCaptchaConnector.class);
+    private static final String SECURED_DESTINATIONS = "/commonauth";
+    private static final String EMAIL_OTP_RECAPTCHA_ENABLE = "EmailOTP.RecaptchaEnable";
+    public static final String EMAIL_OTP_AUTHENTICATOR_NAME = "email-otp-authenticator";
+    public static final String IS_REDIRECT_TO_EMAIL_OTP = "isRedirectToEmailOTP";
+
+    private IdentityGovernanceService identityGovernanceService;
+
+    @Override
+    public void init(IdentityGovernanceService identityGovernanceService) {
+
+        this.identityGovernanceService = identityGovernanceService;
+    }
+
+    @Override
+    public int getPriority() {
+        return 30;
+    }
+
+    @Override
+    public boolean canHandle(ServletRequest servletRequest, ServletResponse servletResponse) throws CaptchaException {
+
+        String path = ((HttpServletRequest) servletRequest).getRequestURI();
+
+        if (StringUtils.isBlank(path) || !(CaptchaUtil.isPathAvailable(path, SECURED_DESTINATIONS))) {
+            return false;
+        }
+
+        if (CaptchaDataHolder.getInstance().isForcefullyEnabledRecaptchaForAllTenants()) {
+            return true;
+        }
+
+        String sessionDataKey = servletRequest.getParameter(FrameworkUtils.SESSION_DATA_KEY);
+        AuthenticationContext context = FrameworkUtils.getAuthenticationContextFromCache(sessionDataKey);
+
+        if (context == null || (EMAIL_OTP_AUTHENTICATOR_NAME.equals(context.getCurrentAuthenticator())
+        && !Boolean.TRUE.equals(context.getProperty(IS_REDIRECT_TO_EMAIL_OTP)))) {
+            return false;
+        }
+
+        if (Boolean.parseBoolean(servletRequest.getParameter("resendCode"))) {
+            return false;
+        }
+
+        return CaptchaUtil.isRecaptchaEnabledForConnector(identityGovernanceService, servletRequest,
+                EMAIL_OTP_RECAPTCHA_ENABLE);
+    }
+
+    @Override
+    public CaptchaPreValidationResponse preValidate(ServletRequest servletRequest, ServletResponse servletResponse) throws CaptchaException {
+
+        CaptchaPreValidationResponse preValidationResponse = new CaptchaPreValidationResponse();
+
+        if (CaptchaDataHolder.getInstance().isForcefullyEnabledRecaptchaForAllTenants()
+                || CaptchaUtil.isRecaptchaEnabledForConnector(
+                        identityGovernanceService, servletRequest, EMAIL_OTP_RECAPTCHA_ENABLE)) {
+
+            preValidationResponse.setCaptchaValidationRequired(true);
+        }
+
+        return preValidationResponse;
+    }
+
+    @Override
+    public CaptchaPostValidationResponse postValidate(ServletRequest servletRequest, ServletResponse servletResponse) throws CaptchaException {
+        return null;
+    }
+
+    public static boolean isEmailRecaptchaEnabled(String tenantDomain) {
```

Comment:
Check if any checks are missing and if we need a method similar to SSOLogin connector